### ISSUE TITLE
error after restart : The password and/or keyfile is wrong (1038)

### ISSUE THREAD

**Issue by stephanBV** [MacPass/MacPass#1225]
**Describe the bug**
I am using a Yubikey to enter the MacPass password (a long hold will always generate the same password), I have been doing this method for awhile now.
I was using MacPass when my MacBook lagged a little so I decided to restart it. Right after I restarted, when I try to get into MacPass I get the error "The password and/or keyhole is wrong (1038).
I tried to update MacPass - same. I transfer the database on another Mac that has MacPass too and I get the same error.
I recreated a new database, using the Yubikey password ans it works, I can unlock the db. 
But my first db with all my password stays  locked.

I don't think I was using a key file, not sure it's been awhile since I set up the db.
Does anyone have any idea how to fix that?

 - MacBook mid-2014 Intel
 - macOS BigSur
 - Version 11.6.4

**Comment by mstarke** [comment]
The nature of the database format is that there is no information on how the key file was derived. Hence you cannot see it you had a key file associated with it or not. One thing you might try is use the remembered keys. If you had this option set to true you can check the preferences and see what files where used as keys. To make security not completely obsolete the database paths are hashed. The potential downsides are explained in the preferences so I think you did not enable it. But it might be worth a try.

**Comment by mstarke** [comment]
To actually check for the data you have to use the terminal:
‘’’’ bash
defaults read com.hicknhacksoftware.macpass RememeberdKeysForDatabases
‘’’’

**Comment by stephanBV** [comment]
Hi Micheal, indeed  I don't think I had that option set to true. I get a "The domain/default pair of (com.hicknhacksoftware.MacPass, RememberKeyFilesForDatabases) does not exist" message when I paste your command in the terminal. 
Before the problem happened I recall I had to restart because I was uninstalling a software called NTFS for Mac I believe and it asked for the Mac to be restarted. MacPass was open when the restart process happened.
MacPass has been refusing the Yubikey static password since.
Also the error code is now 1017, instead of 1038.
Can the file be broken?..

**Comment by stephanBV** [comment]
I SOLVED IT!
It was indeed the keyboard layout used by the Yubikey! I don't know what messed up with it.
It went to System Preferences/Keyboard/Change Keyboard Type.
When it asked me to press the key right next to shift, I hold the Yubikey until I got a message asking to choose a keyboard layout; I chose the US keyboard.
I went back to MacPass and it worked, I could unlock the db.
Thanks Michael for your reply above :)
